### ISSUE TITLE
Delete bin/trial from the tree

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8765]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8765|
|Type|enhancement|
|Created|2016-08-18 09:56:05Z|
|Branch|https://github.com/twisted/twisted/tree/8765-rodrigc-trial-removal|

All the scripts in bin were deleted in [ticket:8491] when they were converted to console scripts
I wanted to delete bin/trial at that time, but faced pushback
because not all the buildbots had been properly converted to use tox.

Now it is time to delete bin/trial from the tree.  It just doesn't work properly from a source checkout
any more:

```
Traceback (most recent call last):
  File "bin/trial", line 21, in <module>
    from twisted.scripts.trial import run
  File "/Users/crodrigues/twisted5/src/twisted/scripts/trial.py", line 16, in <module>
    from twisted.internet import defer
  File "/Users/crodrigues/twisted5/src/twisted/internet/defer.py", line 30, in <module>
    from twisted.logger import Logger
  File "/Users/crodrigues/twisted5/src/twisted/logger/__init__.py", line 89, in <module>
    from ._levels import InvalidLogLevelError, LogLevel
  File "/Users/crodrigues/twisted5/src/twisted/logger/_levels.py", line 9, in <module>
    from constantly import NamedConstant, Names
ImportError: No module named 'constantly'
```

Users should either:
 - type `python setup.py develop` to generate the trial console script
 - do all testing with tox, which installs trial inside the testing virtualenv and then uses that for testing

<details><summary>Searchable metadata</summary>

```
trac-id__8765 8765
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__None None
branch__8765_rodrigc_trial_removal 8765-rodrigc-trial-removal
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1471514165950011 1471514165950011
changetime__1474221881907648 1474221881907648
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/504

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc commented|
|-|-|

Some of the buildbots need to be fixed to not call ./bin/trial in the source tree.

https://github.com/twisted-infra/braid/pull/220

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @rodrigc|
|-|-|

As far as I can see, the Buildbot configuration was updated and we now use tox to run the tests on windows.

so it should be ok to remove it.

I am not happy with the solutions described in the release notes.

At least on Ubunut, `python setup.py develop` will require root ... or using a virtualenv.

And tox is very slow, `$ tox -e py27-alldeps-withcov-posix twisted.words.test.test_xmpproutertap` takes 37 seconds on my system, when the env is already created, while trial from a virtualenv does it in 0.5 seconds.

But I think that we should merge it and look at how to document the Twisted development itself in a separate ticket.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set status to `closed`|
|-|-|

In changeset ca266e967a728f1d0f2f4aebeb966b4ad7af1038
```
#!CommitTicketReference repository="" revision="ca266e967a728f1d0f2f4aebeb966b4ad7af1038"
Merge pull request #504 from twisted/8765-rodrigc-trial-removal

Author: rodrigc
Reviewer: adiroiban
Fixes: #8765

Remove bin/trial from the tree
```
